src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,36 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
+    private Engine engine;
+    private Color color;
+    private double weight;
+    private String model;
+    private boolean isnew;
+
+    private Car(Engine engine, Color color, double weight, String model, boolean isnew) {
+        this.engine = engine;
+        this.color = color;
+        this.weight = weight;
+        this.model = model;
+        this.isnew = isnew;
+    }
+
+    public Car() {
+        engine = new Engine();
+        color = Color.BLUE;
+        weight = 2500;
+        model = "BMV";
+        isnew = true;
+    }
+
+    public Car clone() {
+        Car car = new Car();
+        car.color = color;
+        car.engine = engine;
+        car.model = model;
+        car.weight = weight;
+        car.isnew = isnew;
```

Comment:
i'va used this method to clone this car. My clone will have the same engine object or his own?

Reply:
Don`t sure about it, think the same, but new version Is new!